dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -494,7 +495,15 @@ public void publishAssociated(Contentlet contentlet, boolean isNew, boolean isNe
             throw new DotContentletStateException("Only the working version can be published");
 
         // writes the contentlet object to a file
-        indexAPI.addContentToIndex(contentlet, true, true);
+        HibernateUtil.addCommitListener( new FlushCacheRunnable() {
```

Comment:
You can reduce it to 

 HibernateUtil.addCommitListener(  () -> { 
        try {
                     indexAPI.addContentToIndex(contentlet, true, true);
                 } catch (DotHibernateException e) {
                     Logger.error( this, e.getMessage(), e );
                 }
});